src/app/admin-auth-guard.service.ts

```diff
@@ -0,0 +1,19 @@
+import { Observable } from 'rxjs';
+import { AuthService } from './auth.service';
+import { CanActivate } from '@angular/router';
+import { Injectable } from '@angular/core';
+import { UserService } from './user.service';
+import { map, switchMap } from 'rxjs/operators';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AdminAuthGuard implements CanActivate {
+  constructor(private auth: AuthService, private userService: UserService) {}
+
+  canActivate(): Observable<boolean> {
+    return this.auth.user$
+      .pipe(switchMap(user => this.userService.get(user.uid).valueChanges()))
+      .pipe(map(k => k.isAdmin));
```

Comment:
we should never use abbreviations in our code. They make the code hard to read. Instead of k consider a name that is more descriptive of the variable.